ray-operator/test/e2eautoscaler/raycluster_autoscaler_test.go

```diff
@@ -281,3 +285,78 @@ func TestRayClusterAutoscalerWithDesiredState(t *testing.T) {
 
 	}
 }
+
+func TestRayClusterAutoscalerMinReplicasUpdate(t *testing.T) {
+	for name, tc := range tests {
+
+		test := With(t)
+		g := gomega.NewWithT(t)
+
+		// Create a namespace
+		namespace := test.NewTestNamespace()
+		test.StreamKubeRayOperatorLogs()
+
+		// Script for creating detached actors to trigger autoscaling
+		scriptsAC := newConfigMap(namespace.Name, files(test, "create_detached_actor.py"))
+		scripts, err := test.Client().Core().CoreV1().ConfigMaps(namespace.Name).Apply(test.Ctx(), scriptsAC, TestApplyOptions)
+		g.Expect(err).NotTo(gomega.HaveOccurred())
+		test.T().Logf("Created ConfigMap %s/%s successfully", scripts.Namespace, scripts.Name)
+
+		test.T().Run(name, func(_ *testing.T) {
+			groupName := "test-group"
+
+			rayClusterSpecAC := rayv1ac.RayClusterSpec().
+				WithEnableInTreeAutoscaling(true).
+				WithRayVersion(GetRayVersion()).
+				WithHeadGroupSpec(rayv1ac.HeadGroupSpec().
+					WithRayStartParams(map[string]string{"num-cpus": "0"}).
+					WithTemplate(tc.HeadPodTemplateGetter())).
+				WithWorkerGroupSpecs(rayv1ac.WorkerGroupSpec().
+					WithReplicas(1).
+					WithMinReplicas(0).
+					WithMaxReplicas(5).
+					WithGroupName(groupName).
+					WithRayStartParams(map[string]string{"num-cpus": "1"}).
+					WithTemplate(tc.WorkerPodTemplateGetter()))
+			rayClusterAC := rayv1ac.RayCluster("ray-cluster", namespace.Name).
+				WithSpec(apply(rayClusterSpecAC, mountConfigMap[rayv1ac.RayClusterSpecApplyConfiguration](scripts, "/home/ray/test_scripts")))
+
+			rayCluster, err := test.Client().Ray().RayV1().RayClusters(namespace.Name).Apply(test.Ctx(), rayClusterAC, TestApplyOptions)
+			g.Expect(err).NotTo(gomega.HaveOccurred())
+			test.T().Logf("Created RayCluster %s/%s successfully", rayCluster.Namespace, rayCluster.Name)
+
+			// Wait for RayCluster to become ready
+			g.Eventually(RayCluster(test, rayCluster.Namespace, rayCluster.Name), TestTimeoutMedium).
+				Should(gomega.WithTransform(RayClusterState, gomega.Equal(rayv1.Ready)))
+			g.Expect(GetRayCluster(test, rayCluster.Namespace, rayCluster.Name)).To(gomega.WithTransform(RayClusterDesiredWorkerReplicas, gomega.Equal(int32(1))))
+
+			// Update minReplicas from 0 to 2
+			rayCluster, err = test.Client().Ray().RayV1().RayClusters(namespace.Name).Get(test.Ctx(), rayCluster.Name, metav1.GetOptions{})
+			g.Expect(err).NotTo(gomega.HaveOccurred())
+			rayCluster.Spec.WorkerGroupSpecs[0].MinReplicas = ptr.To(int32(2))
+			rayCluster, err = test.Client().Ray().RayV1().RayClusters(namespace.Name).Update(test.Ctx(), rayCluster, metav1.UpdateOptions{})
+			g.Expect(err).NotTo(gomega.HaveOccurred())
+			test.T().Logf("Updated RayCluster %s/%s successfully", rayCluster.Namespace, rayCluster.Name)
+
+			// Verify that KubeRay creates an additional Pod
+			g.Eventually(RayCluster(test, rayCluster.Namespace, rayCluster.Name), TestTimeoutMedium).
+				Should(gomega.WithTransform(RayClusterDesiredWorkerReplicas, gomega.Equal(int32(2))))
+
```

Comment:
Can you also check if the worker group's `replicas` is equal to 2? Updating replicas from 1 to `max(1, minReplicas)` is the Autoscaler's responsibility.

It is possible that KubeRay creates 2 worker Pods because `minReplicas` is 2, but `replicas` is still 1 because Autoscaler has a bug.

Reply:
@kevin85421 You're correct. I checked with `GetRayClusterWorkerGroupReplicaSum` and found that the replicas field remains at 1, despite minReplicas being set to 2.

![image](https://github.com/user-attachments/assets/846dc38e-dff2-40bb-b7a9-c57670cae964)

Should we open a new issue to track this bug?

Reply:
I find that Autoscaler will only update the `replicas` field only when the next time Autoscaler scales up / down, so `replicas` will not be updated here.

Reply:
> https://github.com/ray-project/kuberay/pull/2634#discussion_r1880845692

this is the expected behavior. We only need to check the field after we submit 5 detached actors to scale up.

Reply:
OK